### PR TITLE
Dfr 3320 notifications judge

### PR DESCRIPTION
**Pull request by hamzahtahirhmcts**
### Jira link
[DFR-3320](https://tools.hmcts.net/jira/browse/DFR-3320)

See [PROJ-XXXXXX](https://tools.hmcts.net/jira/browse/PROJ-XXXXXX)

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java [hunk 1]
```diff
@@ -181,7 +185,7 @@ public void sendAssignToJudgeConfirmationEmailToRespondentSolicitor(FinremCaseDe
      * No Return.
      * <p>Please use @{@link #sendAssignToJudgeConfirmationEmailToIntervenerSolicitor(FinremCaseDetails, SolicitorCaseDataKeysWrapper)}</p>
      *
-     * @param caseDetails instance of CaseDetails
+     * @param caseDetails     instance of CaseDetails
```
Comment (so99y) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java:188]: revert this bit?

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java [hunk 2]
```diff
@@ -558,7 +562,7 @@ public void sendPrepareForHearingEmailIntervener(CaseDetails caseDetails,
 
         NotificationRequest notificationRequest =
             notificationRequestMapper.getNotificationRequestForIntervenerSolicitor(caseDetails,
-            dataKeysWrapper);
+                dataKeysWrapper);
```
Comment (so99y) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java:565]: revert

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java [hunk 3]
```diff
@@ -1907,4 +1911,19 @@ private EmailTemplateNames getIntervenerSendOrderContestedTemplate(IntervenerTyp
             }
         }
     }
+
+    public void sendContestedOrderReadyToReviewToJudge(FinremCaseDetails caseDetails) {
+
+        FinremCaseData caseData = caseDetails.getData();
+        UploadAgreedDraftOrder agreedDraftOrder = caseData.getDraftOrdersWrapper().getUploadAgreedDraftOrder();
+
+        String judgeName = agreedDraftOrder.getJudge();
```
Comment (so99y) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java:1920]: looks like `getJudge` returns Judge's email. It is from the `FixedLists-nonprod.json`
```
  {
    "LiveFrom": "01/01/2017",
    "ID": "FR_fl_AssignToJudge",
    "ListElementCode": "juitestjudgedivorce@gmail.com",
    "ListElement": "JUI Test Judge Divorce"
  }
```

```
  "draftOrdersReviewCollection": [
    {
      "id": "9b606284-5681-4d31-bc46-4261f4c583ac",
      "value": {
        "hearingDate": "2024-11-11",
        "hearingTime": "09:00 AM",
        "hearingType": "First Directions Appointment (FDA)",
        "hearingJudge": "juitestjudgefinrem@mailnesia.com",
```

Comment (so99y) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java:1920]: @jthmcts how can we lookup the judge name from the FixedList in java?

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/draftorders/upload/UploadDraftOrdersSubmittedHandlerTest.java [hunk 4]
```diff
@@ -139,6 +148,30 @@ private static List<SuggestedDraftOrderCollection> suggestedOrdersCollection(
...
+            .build();
+
+        doNothing().when(notificationService).sendContestedOrderReadyToReviewToJudge(any(FinremCaseDetails.class));
+
+        FinremCallbackRequest request = FinremCallbackRequestFactory.from(Long.parseLong(caseReference),
+            CaseType.CONTESTED, caseData);
+
+        var response = uploadDraftOrdersSubmittedHandler.handle(request, AUTH_TOKEN);
+
+        assertThat(response.getConfirmationHeader()).isEqualTo("# Draft orders uploaded");
```
Comment (so99y) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/draftorders/upload/UploadDraftOrdersSubmittedHandlerTest.java:170]: why don't you separate it to another method? or we should change the junit method name.

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/draftorders/upload/UploadDraftOrdersSubmittedHandlerTest.java [hunk 5]
```diff
@@ -139,6 +148,30 @@ private static List<SuggestedDraftOrderCollection> suggestedOrdersCollection(
...
+        assertThat(caseData.getDraftOrdersWrapper().getUploadSuggestedDraftOrder()).isNull();
+        assertThat(caseData.getDraftOrdersWrapper().getUploadAgreedDraftOrder()).isNull();
+    }
```
Comment (so99y) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/draftorders/upload/UploadDraftOrdersSubmittedHandlerTest.java:173]: Should add the following to verify the method was invoked.
```
verify(notificationService).sendContestedOrderReadyToReviewToJudge
```

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java (hunk 1 above)
Comment (hamzahtahirhmcts) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/NotificationService.java:188]: Screenshot 2024-10-23 at 13 04 55
This was an auto style fix. I think it's fine

**Pull request closed without merging**